src/altinn-app-frontend/src/components/GenericComponent.tsx

```diff
@@ -154,6 +155,20 @@ export function GenericComponent<Type extends ComponentExceptGroup>(
     );
   }, [componentValidations]);
 
+  React.useLayoutEffect(() => {
+    if (!hidden && shouldFocus && gridRef.current) {
+      gridRef.current.scrollIntoView();
```

Comment:
Should we perhaps use smooth scrolling? Personally I find it annoying when the page jumps somewhere. Smooth scrolling helps visualize what happens.

There are probably some rules around this though, and I'm not really sure what they are 😅 

@Febakke Do we have any guidelines/rules about automatically scrolling the page?

Reply:
Doesn't really look like it in this piece of code, but this happens as a direct consequence of user action. I would probably have used hash anchors instead, if we didn't already use `window.location.hash` for routing.. :wink: Smooth scrolling is nice (when the animation isn't taking too long, at least), but I think users should be fairly used to the 'clicking this button sends you to the correct part of the document' pattern.

Reply:
Ah, makes sense. Nevermind 🙂